Week10NetworkingCodingExercise/Week10NetworkingCodingExercise/Program.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.IO;
+using System.IO.Compression;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace StreamsIONetworking
+{
+    /// <summary>
+    /// Program runs calculations for querty string.
+    /// </summary>
+    class Program
+    {
+        /// <summary>
+        /// Calls the methods that do the calculations.
+        /// </summary>
+        /// <param name="args"></param>
+        static void Main(string[] args)
+        {
+            //UriDemo();
+            //HttpListenerDemo();
+        }
+
+        /// <summary>
+        /// For understanding how to use query strings.
+        /// </summary>
+        private static void UriDemo()
+        {
+            Console.WriteLine("Uri:");
+            Uri link = new Uri("https://msdn.microsoft.com/en-us/library/system.uri(v=vs.110).aspx?query=abc");
+            Console.WriteLine($"Scheme: {link.Scheme}");
+            Console.WriteLine($"Port: {link.Port}");
+            Console.WriteLine($"Host: {link.Host}");
+            Console.WriteLine($"PathAndQuery: {link.PathAndQuery}");
+
+            Console.WriteLine();
+            Console.WriteLine("UriBuilder:");
+            UriBuilder linkBuilder = new UriBuilder("http://localhost:8001?x=1&y=2");
+            Console.WriteLine($"Scheme: {linkBuilder.Scheme}");
+            Console.WriteLine($"Port: {linkBuilder.Port}");
+            Console.WriteLine($"Host: {linkBuilder.Host}");
+            Console.WriteLine($"Path: {linkBuilder.Path}");
+            Console.WriteLine($"Query: {linkBuilder.Query}");
+
+            Console.ReadKey();
+        }
+
+        /// <summary>
+        /// Runs localhost based on user specified port number.
+        /// </summary>
+        private static void HttpListenerDemo()
+        {
+            Console.WriteLine("Please enter the port number you want to use:");
+            string portNum = Console.ReadLine();
+            int portNumber = Convert.ToInt32(portNum);
+            UriBuilder uri = new UriBuilder("http", "localhost", portNumber);
+
+            Console.WriteLine($"Query: {uri.Query}");
+
+            if (uri.Query != "")
```

Comment:
The uri specified here tells the web server which endpoint to listen on (server and port number) when it starts up. So, this is not where web browser supplied query string values would be provided. Instead, look down at your ProcessRequestAsync method that uses context.Request.QueryString["x"] to get the value specified as "x" in the query string provided by the web browser.